### PR TITLE
migrate to cypress 10

### PR DESCRIPTION
**Pull request by mshima**
Migrate to cypress 10,

- [x] Add '--e2e' flags where we use it. it makes component tests build in, but we only use e2e.
- [x] fix tsconfig for cypress.config.ts 
- [x] fix eslint config for multiples tsconfig files
- [x] fix cypress lint warnings 
- [x] fix jest warning
- [x] workaround cypress coverage missing types due to new typescript config file
- [x] fix eslint webpack config 
- [x] move to new recommended folder cypress/integration -> cypress/e2e
- [x] rename to new recommended extension spec.ts -> cy.ts
- [x] cleanup old files

m1 support has been merged and should be released with 10.2.0 🚀 

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/angular/.eslintrc.json.ejs [hunk 1]
```diff
@@ -35,7 +35,16 @@
   "parserOptions": {
     "ecmaVersion": 2018,
     "sourceType": "module",
-    "project": "./tsconfig.json"
+    "project": [
+      "./tsconfig.app.json",
+<%_ if (cypressTests) { _%>
+      "./<%= CLIENT_TEST_SRC_DIR %>cypress/tsconfig.json",
+<%_ } _%>
+<%_ if (protractorTests) { _%>
```
Comment (Tcharl) [inline generators/client/templates/angular/.eslintrc.json.ejs:43]: Can't move files in their respective folders to ensure consistency?

Comment (mshima) [reply generators/client/templates/angular/.eslintrc.json.ejs:43]: Protractor is deprecated and will be removed.
IMO we could even start to remove it before v8 due to not been maintained for so long. I doubt it’s usable.
Not worth to change anything at this point.

**Review by DanielFran — APPROVED** [review]
Trusting you on this

**Comment by mshima** [comment]
@DanielFran bounty claimed https://opencollective.com/generator-jhipster/expenses/123274

**Comment by DanielFran** [comment]
@mshima approved

**Pull request merged**